src/services/user.service.js

```diff
@@ -0,0 +1,44 @@
+const { data } = require('../data');
+
+const getAll = () => {
+  return data.users;
+};
+
+const getById = (id) => {
+  return data.users.find((user) => user.id === id) || null;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getById(id);
+
+  Object.assign(user, { name });
+
+  return user;
+};
```

Comment:
In the updateUser function, you are not handling the case when the user is not found. If there's no user with the provided id, getById will return null and you will try to update a non-existing object which will result in TypeError. It's a bug in your code. You should add a check to see if the user exists before trying to update it.